### DISCUSSION TITLE
Welcome to OpenSAGE Discussions!

### DISCUSSION THREAD

**Discussion by tgjones** [OpenSAGE/OpenSAGE#571] [General]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we build together 💪.

To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by WinterSnake** [comment]
Does the current network protocol from OpenSAGE interact with regular C&C Generals/Zero Hour installs? I see they are using the same ports but haven't had the time to actually check if they work together.

**Reply by feliwir** [reply]
Hey,

yes they're using the same ports, but apart from that they're not identical/compatible. As discussed in #34 we're going for a lockstep approach aswell, but making the protocols identical is nearly impossible.

**Reply by WinterSnake** [reply]
Thanks for the reply! Shame but I do completely understand - I am looking into it right now and without decompiling the game seems like its not viable.

**Reply by OmniBlade** [reply]
Even with decompiling its nigh impossible to get it to work even if you get the protocol the same because some math results will change depending on compiler options leading to desyncs. Want to take advantage of SSE optimisations? Likely desyncs with the original.

**Reply by feliwir** [reply]
As @OmniBlade pointed out it's pretty hard to get the floating point arithmetics correct over a wide variety of architectures & OSes. Creating something "new" allows to create a more reliable & deterministic for networking

**Reply by WinterSnake** [reply]
Oh yeah no, I totally understand the reasoning. This is definitely the better option.

**Comment by LeFlemard** [comment]
I was watching video about old westwood games and stumbled to open sage, this made me wonder:
Since C&C Renegade actually use the SAGE™, would OpenSAGE actually work to make first person games?
Do not make it a goal, that isn't my intent, was just wondering.

**Reply by OmniBlade** [reply]
C&C Renegade does not use SAGE. Its not clear that the engine that renegade runs on has a name, but its commonly referred to as either the Renegade engine or the W3D engine. The rendering and util libraries from the Renegade engine are used in SAGE, but everything else is new. Given that OpenSAGE is a totally new engine it could be developed in any way the team choose to take it, but currently its to provide similar enough behaviour with the SAGE engine that it can be used as a drop in replacement.
